packages/dos-donts-block/src/DosDontsBlock.tsx

```diff
@@ -1,95 +1,421 @@
+            const placeholderItems: Item[] = [
+                {
+                    id: cuid(),
+                    body: '',
+                    title: '',
+                    type: DoDontType.Do,
+                },
+                {
+                    id: cuid(),
+                    body: '',
+                    title: '',
+                    type: DoDontType.Dont,
+                },
+            ];
+            setAndSaveItems(placeholderItems);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        if (selectedFiles) {
+            setIsUploadLoading(true);
+            uploadFile(selectedFiles);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectedFiles]);
+
+    useEffect(() => {
+        if (doneAll) {
+            batchAddItems(uploadResults);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [doneAll, uploadResults]);
+
+    const addItem = (type: DoDontType) => {
+        const newItems: Item[] = [...localItems, { id: cuid(), body: '', title: '', type }];
+        setAndSaveItems(newItems);
+    };
+
+    const batchAddItems = (assets: Asset[]) => {
+        const newItems: Item[] = [];
+        const assetIds = [];
+        setIsUploadLoading(true);
+
+        for (const image of assets) {
+            newItems.push({
+                id: cuid(),
+                body: '',
+                title: '',
+                imageId: image.id,
+                type: batchUploadType.current,
+            });
+            assetIds.push(image.id);
         }
-        setLocalItems([...updatedItems]);
+        const existingIds = itemImages?.map((x) => x.id) || [];
+        const newIds = [...new Set([...existingIds, ...assetIds])];
+        updateAssetIdsFromKey('itemImages', newIds).then(() => {
+            setAndSaveItems([...localItems, ...newItems]);
+
+            setIsUploadLoading(false);
+        });
     };
 
-    const saveItem = (id: number, value: string, type: 'title' | 'body') => {
-        const updatedItems = localItems;
-        const existingItemIndex = localItems.findIndex((item) => item.id === id);
-        updatedItems[existingItemIndex] = { ...updatedItems[existingItemIndex], [type]: value };
+    const removeItemById = (itemId: string) => {
+        const newItems: Item[] = localItems.filter((item) => item.id !== itemId);
+        setAndSaveItems(newItems);
+    };
+
+    const onChangeItem = (
+        itemId: string,
+        value: string | number | undefined,
+        type: 'title' | 'body' | 'type' | 'imageId'
+    ) => {
+        const newItems: Item[] = localItems.map((item) => (item.id === itemId ? { ...item, [type]: value } : item));
+        setAndSaveItems(newItems);
+    };
 
+    const setAndSaveItems = (newItems: Item[]) => {
+        setLocalItems(newItems);
         setBlockSettings({
-            items: [...updatedItems],
+            ...blockSettings,
```

Comment:
Please remove the `...blockSettings` spread here. Only save the key value pair which actually needs to be updated. Otherwise we have issues on translated pages, as only settings that differ from the main language should be saved to the i18n-settings.